### PR TITLE
Fix typos in docstrings and .rst files that led to rendering errors

### PR DESCRIPTION
**Pull request by jonwzheng**
## Description
This small PR fixes typos that had led to errors when compiling the docs with `sphinx`.  This does not assess the accuracy of any of the docstrings, nor does it add any additional documentation - it just fixes some of the more prominent display issues. Also, it adds some packages to the pip install for the pyproject that were required to locally compile (which I think is what the point of the docs dependencies should be?)

These are still some minor display issues that aren't addressed by this PR, related to AutoAPI duplicating references in the table of contents which messes up the rendering (see [this, for example](https://chemprop.readthedocs.io/en/latest/autoapi/chemprop/nn/loss/index.html#chemprop.nn.loss.EvidentialLoss)). 

## Example

**Before** ([located here](https://chemprop.readthedocs.io/en/latest/autoapi/chemprop/featurizers/atom/index.html#chemprop.featurizers.atom.MultiHotAtomFeaturizer.v1))

Screenshot from 2024-06-04 18-02-27

**After**
Screenshot from 2024-06-04 18-03-11

### CODE REVIEW

File: docs/source/installation.rst [hunk 1]
```diff
@@ -59,10 +59,10 @@ Option 4: Installing via Docker
 -------------------------------
 
 Chemprop can also be installed with Docker, making it possible to isolate the Chemprop code and environment.
-To install and run Chemprop in a Docker container, first install `Docker`_.
+To install and run Chemprop in a Docker container, first `install Docker`_.
```
Comment (JacksonBurns) [inline docs/source/installation.rst:62]: Why was this required?

**Review by JacksonBurns — COMMENTED** [review]
Quick question!

File: docs/source/installation.rst (hunk 1 above)
Comment (jonwzheng) [reply docs/source/installation.rst:62]: Previously in this file (line 6), the `Docker` reference is defined and points to the Docker "Get Started" page, whereas here we want it to point to the Docker "Get Docker" page. I think the easiest way to accommodate was just to make these two separate references.

Comment (JacksonBurns) [reply docs/source/installation.rst:62]: Thanks for catching that! I'm surprised that wasn't an error.

Comment (jonwzheng) [reply docs/source/installation.rst:62]: It does give an error, but one that's recoverable, see the [current install page ](https://chemprop.readthedocs.io/en/latest/installation.html) and how the first \`Docker\` ref is messed up.
I caught these errors because I parsed through the error log during compilation - usually this isn't seen because the docs are compiled automatically, but I was compiling on my local device.

**Review by JacksonBurns — COMMENTED** [review]
I've not rendered these locally myself, but I trust you that the formatting errors are fixed. The changes look good to me.

The CI is failing on the formatting step - I think `black` doesn't like line 59 in `splitting.py` because it has some spaces in it and line 200 in `model.py` for the same reason.

**Comment by jonwzheng** [comment]
okay, I'll try removing the extra lines and see if it still compiles fine. Sorry about that!
BTW (and this is something I recently learned):  you can see the RTD builds for PRs on https://readthedocs.org/projects/chemprop/builds/ - so the docs for this PR is https://chemprop--901.org.readthedocs.build/en/901/

**Comment by jonwzheng** [comment]
ran it through black and made sure everything still renders nicely. Should be good now!

**Review by JacksonBurns — APPROVED** [review]
@KnathanM please review this as well!

**Review by KnathanM — APPROVED** [review]

File: chemprop/models/model.py [hunk 2]
```diff
@@ -196,11 +196,12 @@ def predict_step(self, batch: TrainingBatch, batch_idx: int, dataloader_idx: int
...
-            molecules/reactions, ``t`` is the number of tasks, and ``s`` is the number of targets
-            per task. The final dimension is flattened, so that the targets for each task are
-            grouped. I.e., the first ``t`` elements are the first target for each task, the second
-            ``t`` elements the second target, etc.
+            * | regression/binary classification: ``n x (t * s)``, where ``n`` is the number of input
+              | molecules/reactions, ``t`` is the number of tasks, and ``s`` is the number of targets
+              | per task. The final dimension is flattened, so that the targets for each task are
+              | grouped. I.e., the first ``t`` elements are the first target for each task, the second
+              | ``t`` elements the second target, etc.
+
```
Comment (jonwzheng) [inline chemprop/models/model.py:204]: I know we just approved this, but I think the "|" weren't necessary (the space addition after the paragraph, however, is.)

**Review by jonwzheng — COMMENTED** [review]
IK we just approved everything but quick comment before we merge in / potential change

**Review by JacksonBurns — APPROVED** [review]
